src/main/java/picocli/CommandLine.java

```diff
@@ -6693,6 +6705,7 @@ void initDefaultValueProvider(Class<? extends IDefaultValueProvider> value, IFac
             void updateHelpCommand(boolean value)           { if (isNonDefault(value, DEFAULT_IS_HELP_COMMAND))             {isHelpCommand = value;} }
             void updateSubcommandsRepeatable(boolean value) { if (isNonDefault(value, DEFAULT_SUBCOMMANDS_REPEATABLE))      {subcommandsRepeatable = value;} }
             void updateAddMethodSubcommands(boolean value)  { if (isNonDefault(value, DEFAULT_IS_ADD_METHOD_SUBCOMMANDS))   {isAddMethodSubcommands = value;} }
+            void updateHidden(boolean value)                { if (isNonDefault(value, DEFAULT_HIDDEN)) { isHidden = value; } }
```

Comment:
@remkop I didn't see another way of access the `hidden` prop, but I'm probably missing something as it is used for the help text output.
If you have any alternative idea let me know, and I could remove this method (and revert the MINOR version bump)

Reply:
@bdemers The `@Command(hidden)` value is captured as a property of the command's `UsageHelpMessageSpec`. I admit this is not easy to find. Perhaps the javadoc for `picocli.CommandLine.Command#hidden` should have a `@see picocli.CommandLine.Model.UsageMessageSpec#hidden()` reference to make this a bit easier to find. Can you do that as part of the PR?

The `Help` class has its own map of subcommands, and when this map is built up in `CommandLine.Help#addAllSubcommands`, hidden subcommands are excluded. This means that the logic for `Help` is simpler, because it does not need to care about hidden subcommands any more, but can be a bit puzzling if you miss this. This should be mentioned in the javadoc for `Help#addAllSubcommands` and the javadoc for `Help#subcommands`. Can you include that in the PR?

Good finds, thanks! 👍

Reply:
Since we can use `picocli.CommandLine.Model.UsageMessageSpec#hidden`, all changes to `CommandLine` can be reverted.